test/entrypoint_test.go

```diff
@@ -43,30 +40,21 @@ func TestEntrypointRunningStepsInOrder(t *testing.T) {
 	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
 	defer tearDown(t, c, namespace)
 
-	t.Logf("Creating Task and TaskRun in namespace %s", namespace)
-	task := &v1beta1.Task{
-		ObjectMeta: metav1.ObjectMeta{Name: epTaskName, Namespace: namespace},
-		Spec: v1beta1.TaskSpec{
-			Steps: []v1beta1.Step{{Container: corev1.Container{
-				Image: "ubuntu",
-				Args:  []string{"-c", "sleep 3 && touch foo"},
-			}}, {Container: corev1.Container{
-				Image: "ubuntu",
-				Args:  []string{"-c", "ls", "foo"},
-			}}},
-		},
-	}
-	if _, err := c.TaskClient.Create(task); err != nil {
-		t.Fatalf("Failed to create Task: %s", err)
-	}
-	taskRun := &v1beta1.TaskRun{
+	t.Logf("Creating TaskRun in namespace %s", namespace)
+	if _, err := c.TaskRunClient.Create(&v1beta1.TaskRun{
 		ObjectMeta: metav1.ObjectMeta{Name: epTaskRunName, Namespace: namespace},
 		Spec: v1beta1.TaskRunSpec{
-			TaskRef:            &v1beta1.TaskRef{Name: epTaskName},
-			ServiceAccountName: "default",
+			TaskSpec: &v1beta1.TaskSpec{
+				Steps: []v1beta1.Step{{
+					Container: corev1.Container{Image: "busybox"},
+					Script:    "sleep 3 && touch foo",
```

Comment:
Feel free to leave since you're not responsible, but do we know why the sleep 3 is here? It seems like it would have no effect aside from slowing tests down.

Reply:
If steps are running "at the same time", it's possible this would actually end up touching foo slightly before the other container tries to read it, which could lead to false positives. By sleeping a little we reduce (but admittedly not entirely eliminate) the possibility.